chainerx_cc/chainerx/routines/loss.cc

```diff
@@ -24,4 +28,19 @@ Array HuberLoss(const Array& x1, const Array& x2, Scalar delta) {
     return Where(abs_a < delta_array, 0.5 * Square(a), delta * (abs_a - Scalar{0.5} * delta));
 }
 
+Array Hinge(const Array& x1, const Array& x2, const std::string& norm) {
```

Comment:
Please raise `DimensionError` when `ndim` and `shape` of `x1` or `x2` are wrong.

Reply:
`norm` should not be passed as a string value. How about
1. defining an enum class or
2. passing as a float value?

Reply:
Could I know why `string` is not preferred?
If defined as `float`, what could be the values of `norm` corresponding to `L1` and `L2` ?

Reply:
`string` is not preferred because of the performance.

Reply:
> Could I know why `string` is not preferred?
> If defined as `float`, what could be the values of `norm` corresponding to `L1` and `L2` ?

Thanks.
Could you please also answer the second question?

Reply:
I mean, let's support general L-p norm. In this case, `float norm` represents the p.

Reply:
Oh, I get it now. So, why not define it as an `int` since 1 and 2 are integer values?

Reply:
Do you mean to say that we can allow `p` value to be non integers as well to make it general?

Reply:
Yes. L-p norm is defined for all real p >= 1.